helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -947,6 +947,10 @@ public boolean isMaintenanceModeEnabled() {
     return _isMaintenanceModeEnabled;
   }
 
+  public void enableMaintenanceMode() {
```

Comment:
The logic is fine. But let's be more careful here. This method is not supposed to be used anywhere else except when we update the ZK M mode signal. Let's comment on it here in the code. Also, please add a TODO here that we need to separate read-only cache from the updatable cache in the near future.